src/connectedvmware/azext_connectedvmware/custom.py

```diff
@@ -236,7 +236,29 @@ def show_inventory_item(
     inventory_item
 ):
 
-    return client.get(resource_group_name, vcenter.split('/')[-1], inventory_item.split('/')[-1])
+    morefId: str = inventory_item.split('/')[-1]
+    moType = morefId.rsplit('-', 1)[0]
+    inventoryType = 'InventoryItem'
+
+    if moType == 'datastore':
+        inventoryType = 'DatastoreInventoryItem'
+    elif moType == 'domain':
+        inventoryType = 'ClusterInventoryItem'
+    elif moType == 'dvportgroup':
+        inventoryType = 'VirtualNetworkInventoryItem'
+    elif moType == 'host':
+        inventoryType = 'HostInventoryItem'
+    elif moType == 'network':
+        inventoryType = 'VirtualNetworkInventoryItem'
+    elif moType == 'resgroup':
+        inventoryType = 'ResourcePoolInventoryItem'
+    elif moType == 'vm':
+        inventoryType = 'VirtualMachineInventoryItem'
+    elif moType == 'vmtpl-vm':
+        inventoryType = 'VirtualMachineTemplateInventoryItem'
```

Comment:
Avoid using morefId to inventoryType conversion in the sdk.
InventoryType property is already populated by the service, we should rely on that to deserialize to the corresponding inventory model. Check InventoryItemPropertiesConverter in service.

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/connectedvmware/resource-manager/Microsoft.ConnectedVMwarevSphere/stable/2023-10-01/connectedvmware.json#L5457

Reply:
Using `resource` client to fetch the inventory item, which shows all the props as seen using `az resource show` command. Setting custom deserializer is not straightforward, we can look into that later, if required.